### PR TITLE
Fixed server table's quick search and filter for multiple profiles

### PR DESCRIPTION
**Pull request by rimashah25**
Fixes: #7049 and 
Fixes: #7052 

<hr/>

## Which Traffic Control components are affected by this PR?

- Traffic Portal

## What is the best way to verify this PR?

Add servers with multiple profiles and quick search and filter for any profile that is not a primary profile

## If this is a bugfix, which Traffic Control versions contained the bug?

7.0.1 (master)

## PR submission checklist
- [ ] This PR has no tests as it only changes the display of a table column (UI).  
- [ ] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

**Review by ocket8888 — COMMENTED** [review]
This fixes the search and appears to render fine, but some other tables aren't displaying Profiles for servers properly, e.g. the Delivery Service server-assignment table pictured below.

image

Also, #7049 specifically mentions viewing servers filtered by Profile via the "View Servers" menu option on a Profile's page. This PR doesn't fix that.

Also, the table column name is currently still 'Profile', should probably now be 'Profiles' or perhaps 'Profile(s)'. But that's less important.

**Comment by rimashah25** [comment]
> This fixes the search and appears to render fine, but some other tables aren't displaying Profiles for servers properly, e.g. the Delivery Service server-assignment table pictured below.
> 
> image
> 
> Also, #7049 specifically mentions viewing servers filtered by Profile via the "View Servers" menu option on a Profile's page. This PR doesn't fix that.
> 
> Also, the table column name is currently still 'Profile', should probably now be 'Profiles' or perhaps 'Profile(s)'. But that's less important.

Resolved #7049 (filtered by Profile using column Filter).

File: traffic_portal/app/src/common/modules/table/agGrid/CommonGridController.js [hunk 1]
```diff
@@ -301,6 +301,34 @@ let CommonGridController = function ($scope, $document, $state, userModel, dateU
...
+				this.columns[i].filterParams = {
+					textCustomComparator: (filter, value, filterText) => {
+						const filterTextLowerCase = filterText.toLowerCase();
+						const valueLowerCase = value.toString().toLowerCase();
+						const profileNameValue = valueLowerCase.split(",");
+						switch (filter) {
+							case 'contains':
+								return valueLowerCase.indexOf(filterTextLowerCase) >= 0;
+							case 'notContains':
+								return valueLowerCase.indexOf(filterTextLowerCase) === -1;
```
[the remark this answers is not in the document]
Comment (rimashah25) [reply traffic_portal/app/src/common/modules/table/agGrid/CommonGridController.js:315]: So, you want me to use `includes` for for `contains`, `notContains`, `startsWith`, `endsWith`?

File: traffic_portal/app/src/common/modules/table/agGrid/CommonGridController.js [hunk 2]
```diff
@@ -301,6 +301,34 @@ let CommonGridController = function ($scope, $document, $state, userModel, dateU
...
+							case 'equals':
+								return profileNameValue.includes(filterTextLowerCase);
+							case 'notEqual':
+								return !profileNameValue.includes(filterTextLowerCase);
+							case 'startsWith':
+								return valueLowerCase.indexOf(filterTextLowerCase) === 0;
+							case 'endsWith':
+								var index = valueLowerCase.lastIndexOf(filterTextLowerCase);
+								return index >= 0 && index === (valueLowerCase.length - filterTextLowerCase.length);
```
[the remark this answers is not in the document]
Comment (rimashah25) [reply traffic_portal/app/src/common/modules/table/agGrid/CommonGridController.js:324]: Also though, please don't use var ==> Resolved.

File: traffic_portal/app/src/common/modules/table/agGrid/CommonGridController.js [hunk 3]
```diff
@@ -301,6 +301,34 @@ let CommonGridController = function ($scope, $document, $state, userModel, dateU
...
+							case 'contains':
+								return valueLowerCase.indexOf(filterTextLowerCase) >= 0;
+							case 'notContains':
+								return valueLowerCase.indexOf(filterTextLowerCase) === -1;
+							case 'equals':
+								return profileNameValue.includes(filterTextLowerCase);
+							case 'notEqual':
+								return !profileNameValue.includes(filterTextLowerCase);
+							case 'startsWith':
+								return valueLowerCase.indexOf(filterTextLowerCase) === 0;
```
[the remark this answers is not in the document]
Comment (rimashah25) [reply traffic_portal/app/src/common/modules/table/agGrid/CommonGridController.js:321]: Spoke to the reviewer and current (and below) comments are more for knowledge purpose and doesn't require any change.

**Review by ocket8888 — APPROVED** [review]
Works

**Pull request merged**